File: cpp/LinkedList/insertionatend.cpp
#include<bits/stdc++.h>
using namespace std;
class Node{
    public:
    int data;
    Node*next;
    Node(int d,Node* next1){
        data=d;
        next=next1;
    }
    Node(int d){
        data=d;
        next=NULL;
    }
};

Node* insert(Node* &head,int d)
{
    if(head==NULL)
    {
        head=new Node(d);
        return head;
    }
    Node* temp=head;
    while(temp->next!=NULL){
        temp=temp->next;
    }
    temp->next=new Node(d);
    return head;
}

void print(Node*head)
{
    Node* temp=head;
    while(temp!=NULL)
    {
        cout<<temp->data<<" ";
        temp=temp->next;
    }
}

int main()
{
    Node*head=NULL;
    int n,d,x;
    cin>>n;
    vector<int>arr;

    for(int i=0;i<n;i++)
    {
        cin>>d;
        insert(head,d);
    }
    cin>>x;
    insert(head,x);

    print(head);
    return 0;
}

File: python/LinkedList/insertionatend.py
class Node:
    def __init__(self, data, next1=None):
        self.data = data
        self.next = next1
        
class LinkedList:
    def __init__(self):
        self.head = None
        
    def insert(self,data):
        if self.head is None:
            self.head=Node(data)
            return
        
        temp=self.head
        while(temp.next):
            temp=temp.next
            
        temp.next=Node(data)
        
    def printlist(self):
        temp=self.head
        while(temp):
            print(temp.data)
            temp=temp.next
            
            
if __name__=='__main__':
    linked_list=LinkedList()
    n=int(input())
    
    elements=list(map(int,input().split()))
    for data in elements:
        linked_list.insert(data)
        
    x=int(input())
    linked_list.insert(x)
    linked_list.printlist()
        
        